File: Templates/Project Templates/macOS/Application Extension/Audio Unit Extension.xctemplate/AudioUnitViewController.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"
#import "___PACKAGENAMEASIDENTIFIER___AudioUnit.h"

@interface ___FILEBASENAME___ ()

@end

@implementation ___FILEBASENAME___ {
    AUAudioUnit *audioUnit;
}

- (void) viewDidLoad {
    [super viewDidLoad];
    
    if (!audioUnit) {
        return;
    }
    
    // Get the parameter tree and add observers for any parameters that the UI needs to keep in sync with the AudioUnit
}

- (AUAudioUnit *)createAudioUnitWithComponentDescription:(AudioComponentDescription)desc error:(NSError **)error {
    audioUnit = [[___PACKAGENAMEASIDENTIFIER___AudioUnit alloc] initWithComponentDescription:desc error:error];
    
    return audioUnit;
}

@end

File: Templates/Project Templates/macOS/Application Extension/Audio Unit Extension.xctemplate/AudioUnitViewController.swift
//___FILEHEADER___

import CoreAudioKit

public class ___FILEBASENAME___: AUViewController, AUAudioUnitFactory {
    var audioUnit: AUAudioUnit?
    
    public override func viewDidLoad() {
        super.viewDidLoad()
        
        if audioUnit == nil {
            return
        }
        
        // Get the parameter tree and add observers for any parameters that the UI needs to keep in sync with the AudioUnit
    }
    
    public func createAudioUnit(with componentDescription: AudioComponentDescription) throws -> AUAudioUnit {
        audioUnit = try ___PACKAGENAMEASIDENTIFIER___AudioUnit(componentDescription: componentDescription, options: [])
        
        return audioUnit!
    }
    
}
